File: app/src/main/java/com/example/chi_12_di/data/repository/PhotosRepositoryImplModule.java
package com.example.chi_12_di.data.repository;

import com.example.chi_12_di.domain.repository.IPhotosRepository;
import dagger.Module;
import dagger.Provides;

@Module
public class PhotosRepositoryImplModule {

    @Provides
    IPhotosRepository providePhotosRepository (PhotosRepositoryImpl photosRepository){
        return photosRepository;
    }
}

File: app/src/main/java/com/example/chi_12_di/di/PhotosRepositoryImplModule.kt
package com.example.chi_12_di.di

import com.example.chi_12_di.data.repository.PhotosRepositoryImpl
import com.example.chi_12_di.domain.repository.IPhotosRepository
import dagger.Binds
import dagger.Module

@Module
interface PhotosRepositoryImplModule {

    @Binds
    fun providePhotosRepository(photosRepository: PhotosRepositoryImpl): IPhotosRepository
}
